### PR TITLE
Updated pillow, unlocked test

### PR DESCRIPTION
**Pull request by bsekachev**
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: cvat/apps/engine/media_extractors.py [hunk 1]
```diff
@@ -658,7 +658,7 @@ def save_as_chunk(self, images, chunk_path):
                         if has_exif_rotation(pil_image):
                             rot_image = ImageOps.exif_transpose(pil_image)
                             try:
-                                if rot_image.format == 'TIFF':
+                                if pil_image.format == 'TIFF':
```
Comment (bsekachev) [inline cvat/apps/engine/media_extractors.py:661]: rot_image.format is None after exif_transpose 
It results to jpeg image saved with .tif filename

File: cvat/apps/engine/media_extractors.py [hunk 2]
```diff
@@ -663,7 +663,7 @@ def save_as_chunk(self, images: Iterable[tuple[Image.Image|io.IOBase|str, str, s
                     if has_exif_rotation(image):
                         rot_image = ImageOps.exif_transpose(image)
                         try:
-                            if rot_image.format == 'TIFF':
+                            if image.format == 'TIFF':
```
Comment (SpecLad) [inline cvat/apps/engine/media_extractors.py:666]: I assume this is a bugfix? If so, please add a changelog entry.

File: changelog.d/20231227_100059_boris_update_pillow.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+### Fixed
+
+- TIFF images are saved with .jpeg extension in original chunks
+  (<https://github.com/opencv/cvat/pull/7100>)
```
Comment (bsekachev) [inline changelog.d/20231227_100059_boris_update_pillow.md:4]:
```suggestion
- TIFF images are saved with .jpeg extension in original chunks
  (<https://github.com/opencv/cvat/pull/7100>)
- EXIF rotated TIFF images are handler incorrectly
  (<https://github.com/opencv/cvat/pull/7100>)
```

**Review by azhavoro — APPROVED** [review]

File: changelog.d/20231227_100059_boris_update_pillow.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+- EXIF rotated TIFF images are handler incorrectly
```
Comment (bsekachev) [inline changelog.d/20231227_100059_boris_update_pillow.md:5]:
```suggestion
- EXIF rotated TIFF images are handled incorrectly
```

File: changelog.d/20231227_100059_boris_update_pillow.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+### Fixed
+
+- TIFF images are saved with .jpeg extension in original chunks
```
Comment (bsekachev) [inline changelog.d/20231227_100059_boris_update_pillow.md:3]:
```suggestion
- `TIFF` images are saved as `JPEG` images with `.tif` extension in original chunks
```

**Pull request merged**